flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/nodes/datastream/DataStreamCorrelate.scala

```diff
@@ -36,14 +36,14 @@ import org.apache.flink.types.Row
 class DataStreamCorrelate(
     cluster: RelOptCluster,
     traitSet: RelTraitSet,
-    inputNode: RelNode,
```

Comment:
hi, sorry for missing any commons about this fix. After decoration phase, the class type of inputNode is `HepRelVertex` and it will throws ClassCastException at `val inputDS = inputNode.asInstanceOf[DataStreamRel].translateToPlan(tableEnv)`. You can reproduce this exception by running tests in `DataStreamUserDefinedFunctionITCase`. The reason why there is no problems after runVolcanoPlanner is that `DataStreamCorrelateRule` does the transformation from `RelSubset` to `DataStreamRel`. I think it's better to override the `input`  parameter and use `getInput` when translate to plan. What do you think, thx~